online-shop/src/app/edit-form/edit-form.component.ts

```diff
@@ -0,0 +1,56 @@
+import { Component, OnInit, Input } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+import { ActivatedRoute } from '@angular/router';
+import { Post } from '../product/post';
+import { FormBuilder, FormControl } from '@angular/forms';
+import { Validators } from '@angular/forms';
+
+@Component({
+  selector: 'app-edit-form',
+  templateUrl: './edit-form.component.html',
+  styleUrls: ['./edit-form.component.css'],
+})
+export class EditFormComponent implements OnInit {
+  product: Post = {
+    id: 0,
+    name: '',
+    category: '',
+    image: '',
+    price: 0,
+    description: '',
+  };
+
+  id = this.route.snapshot.paramMap.get('id');
```

Comment:
**suggestion:** you should have this assignment in the component lifecycle, more specifically in `ngOnInit`